lighthouse-service/event_handler/start_evaluation_handler.go

```diff
@@ -143,8 +144,17 @@ func (eh *StartEvaluationHandler) sendEvaluationFinishedWithErrorEvent(start, en
 }
 
 func getEvaluationTimestamps(e *keptnv2.EvaluationTriggeredEventData) (string, string, error) {
-	if e.Evaluation.Start != "" && e.Evaluation.End != "" {
-		return e.Evaluation.Start, e.Evaluation.End, nil
+	if (e.Evaluation.Start != "" && e.Evaluation.End != "") || (e.Evaluation.Timeframe != "") {
+		params := timeutils.GetStartEndTimeParams{
+			StartDate: e.Evaluation.Start,
+			EndDate:   e.Evaluation.End,
+			Timeframe: e.Evaluation.Timeframe,
+		}
+		start, end, err := timeutils.GetStartEndTime(params)
+		if err != nil {
+			return "", "", err
+		}
```

Comment:
What happens when timeframe is set, but start and end are not set?

I think in issue #4079 I got it wrong, what I meant was that
`end = evaluation.triggered.time`, and `start = end - timeframe`

Reply:
when only the time frame is set, we adapt the same logic we did previously in the evaluation endpoint: `end = time.Now()`, `start = time.Now() - timeframe`

Reply:
The big difference here is that the evaluation endpoint is called in a specific moment.

The cloudevent is sent at a specific moment, but the time it is retrieved by lighthouse could be a different time.
Hence I would propose to not use `time.Now()`, but the actual cloudevents triggered time (when was the evaluation triggered). 

I know it doesn't make much difference in our current setup (maybe a couple of milliseconds), but using the cloudevents time would make the behaviour reproducible and deterministic.
Think about re-triggering an evaluation.